blocks/cards/cards.js

```diff
@@ -1 +1,88 @@
-/* TODO */
+import {
+  parseFragment,
+  removeEmptyElements,
+  render,
+  convertExcelDate,
+  timeSince,
+} from '../../scripts/scripts.js';
+
+let cardData;
+
+async function getCardData() {
+  const response = await fetch('/blocks/cards/mockData.json');
+  cardData = await response.json();
+}
+
+const prependImage = 'https://main--helix-sportsmagazine--headwirecom.hlx.live/';
+
+export default async function decorate(block) {
+  if (!cardData) {
+    await getCardData();
+  }
+  const cardLinks = [...block.querySelectorAll('p>a[href]')].map((element) => element.href);
+
+  const gdPlusCards = block.classList.contains('gd');
+
+  const cardBlocks = [...document.querySelectorAll(".cards.block[data-block-name='cards']")];
+  const indexInPage = cardBlocks.findIndex((element) => element.isEqualNode(block));
+  const reverse = !(indexInPage === 0 || indexInPage % 2 === 0);
+
+  const cardList = cardLinks.map((cardLink) => {
+    const cardSearchQuery = cardLink.split('/').slice(3).join('/');
+    const cardObj = cardData.data.find((obj) => obj.path === cardSearchQuery);
+    if (cardObj) {
+      cardObj.href = cardLink;
+      cardObj.image = prependImage + cardObj.image;
+    }
+    return cardObj;
+  });
+
+  const mainCard = cardList[0];
+
+  const HTML_TEMPLATE = `
+  <div class="card-block-wrapper ${reverse ? 'reverse' : ''}">
+    <a class="main-card" href="${mainCard.href}">
+      <div class="image-bg">
+        <img loading="lazy" src="${mainCard.image}" alt="${mainCard.imageAlt}"/>
```

Comment:
@DriftingSands can you use `<picture>` instead using this function https://github.com/headwirecom/helix-sportsmagazine/blob/main/scripts/lib-franklin.js#L450 to create an optimized image for cards ?